src/tls.c

```diff
@@ -2117,6 +2131,23 @@ static int TLSX_SNI_Parse(WOLFSSL* ssl, const byte* input, word16 length,
     matched = cacheOnly || (XSTRLEN(sni->data.host_name) == size &&
          XSTRNCMP(sni->data.host_name, (const char*)input + offset, size) == 0);
 
+#if defined(HAVE_ECH)
+    if (!matched && ech != NULL) {
+        workingConfig = ech->echConfig;
+
+        while (workingConfig != NULL) {
+            matched = XSTRLEN(workingConfig->publicName) == size &&
```

Comment:
the statement doesn't end after XSTRLEN, it checks if it == size and if  XSTRNCMP == 0 which are boolean returns, does that need to be cast to byte?

Reply:
I see, thanks @jpbland1, when scanning the code I had missed the '==' and that it is a condition assigning 1/0 to matched. My preference is that there is still a cast as @dgarske mentions and maybe breaking up that statement for maintaining/readability. Would perfer the typical `if (....) { matched = 1;}`. Marking this as resolved though because for me it's just preference.